### ISSUE TITLE
Including a possibility to have a "prediction" of the price of ETH in the calculations.

### ISSUE THREAD

**Issue by mathvdh** [AgeManning/EthereumMiningCalculator#60]
So, I know predicting the price of ETH is difficult if not impossible.

This being said, I think we can see a slow linear growth (in general) in the price.

So I was wondering if it wouldn't be interesting to give an option to activate a 'predicted' price for ETH instead of a fixed price ...

Does it make sense ? (Are my thoughts at least understandable to you guys ?)

Cheers

**Comment by AgeManning** [comment]
Yeah perfect. 
I specifically stayed away from this, as building something to predict eth price could get controversial. 

But I think you are right. I could (without too much difficulty) just implement the same logic used for difficulty, (i.e sample some points and fit a curve) and give users the option to use this. 

There are a few small concerns I'll have to think about. 
1. Will it be too confusing for the user having all these predictions and what not happening. 
2. The costs tables, ie. how much do you make per hour/ year etc will no longer be really valid as it depends on which part of the year (with your predicted price) 
3. The expected return graphs could have some weird cases where you don't make profit but then later you do and predicting an ROI could be difficult. 

If you think its a useful addition, i'll add it in. 

Thanks for the input and feedback :).

**Comment by mathvdh** [comment]
Well, I think it's useful.

I think a compromise would be to put it in.

But then deactivate it by default in the "basic" and "custom" options and give an option to activate it only in the "Advanced" option (Maybe with a disclaimer : "This is experimental and we can't possibly predict the price of ETH correctly ... use at your own risk ... blabla")

What do you think ?

**Comment by AgeManning** [comment]
Yep great idea!. 
I'd just have to fetch data from an API like Polo or something, then could just feed it straight into the predictive-service you were looking at. 

Do you think it would be useful to allow users to edit the coefficients of the form, like in the predictive difficulty?. So for example they could make their own line or function to see how their predictions perform?

**Comment by mathvdh** [comment]
Yes I think it would be useful for advanced users. On the other end users who are that advanced can do their own predictions using R or whatever :)

OT : I see on your picture you have a helmet .. What kind of Bike do you have ??? I have Honda Hornet 600 :)

**Comment by AgeManning** [comment]
Yeah cool. I'll add it in then. 

I ride a cbr600rr. The guy who is helping me build this, Paul also rides. He had an R1 but has since moved on to more adventure bike kind of things. 

To introduce us a bit better, I'm like kinda the math guy, (all the ugly code you see, is probably my doing ;) ) and Paul is the everything else guy. He makes the math and things look pretty and function. He'll be watching the thread, but hasn't responded because you asked maths questions :) .  @paulhauner

**Comment by mathvdh** [comment]
Hehe do you know the Hornet 600 as the same engine (a bit tuned-down) as the CRB 600RR !

Good job to both of you anyways :) (Math or Visuals)

Personally I'm kind of an all-around guy (I started my career doing machine learning research and a PHD which I never finished :) Then I ended up doing websites and mobile apps and now doing network engineering and big data stuff ...

**Comment by AgeManning** [comment]
Yeah cool. Too bad we're not in the same country, can't go for rides :(. 
 I've done a bit of AI stuff too in my engineering side hobby thing. I'm also a fairly all-arounder but terrible with graphics design and making UI look pretty. 
I've worked a little bit with big data, mainly programming SPROCS in SQL to predict betting things, from a mathematics/statistics point of view. 
Formally, I'm a particle physicist and should be finishing my PhD in a few months. This calculator is Paul and I's kind of procrastination from the things we should otherwise be doing. 
I'm also a crypto nerd, love cryptocurrencies, well blockchain stuff. Ethereum mainly as you can probably guess.
